### ISSUE TITLE
Main menu not displayed when logged into customer account

### ISSUE THREAD

**Issue by florine2623** [PrestaShop/PrestaShop#26921]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When I choose a specific Group access + Default Customer Group while creating a new Customer Group; when add a Customer to it, in the FO when the customer logs into his account, the main menu disappears.

Logged out :
<img width="1137" alt="Screenshot 2021-12-08 at 15 36 22" src="https://user-images.githubusercontent.com/16019289/145226897-87d1e911-5079-448c-8b82-83ecb6329f58.png">

Logged in :
<img width="1141" alt="Screenshot 2021-12-08 at 15 36 44" src="https://user-images.githubusercontent.com/16019289/145226911-d72bcf29-abb6-4000-a430-d63066f5fdb8.png">

### Expected behavior

The main menu should be displayed

### Steps to reproduce

1. Go to BO > Shop Parameters > Customer Settings > Groups
2. Add new Group > Choose a name > Save
3. Click Edit on the newly created group, make sure Module Main Menu is enabled :
<img width="221" alt="Screenshot 2021-12-08 at 15 39 15" src="https://user-images.githubusercontent.com/16019289/145227347-8597257a-f1a2-4bac-8beb-fc900741e006.png">

4. Go to Customers > Customers > Add new customer
5. Fill the form, there should be only one group access : the same as the default customer group. e.g : 
<img width="867" alt="Screenshot 2021-12-08 at 15 35 15" src="https://user-images.githubusercontent.com/16019289/145227800-13e04533-3583-46af-b232-e4cecc8a6a96.png">

6. Go to FO > Log into your newly created customer account 
7.  See that the Main menu is not displayed

### PrestaShop version(s) where the bug happened

1.7.7.x, 1.7.8.x, develop

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

ps_mainmenu

**Comment by leemyongpakvn** [comment]
@florine2623 Indeed it is not a bug, you just need one more setting - check Group access in Back-Office > Catalog > Categories > the desired category. 
![BackOffice_Catalog_Categories_GroupAccess](https://user-images.githubusercontent.com/3759923/181284449-2159532b-7541-4bfb-840f-4ef8440cdda9.png)

**Comment by sitte** [comment]
It adds all id_category and id_group combinations that does not allready exists. Make backup before You use it. 

```
INSERT INTO ps_category_group (id_category, id_group)
SELECT c.id_category, g.id_group
FROM ps_category AS c
CROSS JOIN ps_group AS g
LEFT JOIN ps_category_group AS cg
  ON cg.id_category = c.id_category AND cg.id_group = g.id_group
WHERE cg.id_category IS NULL;
```

**Comment by boo-code** [comment]
Still reproduces on 9.2.x, and it is a symptom of #27674 rather than a separate bug - which also explains why @leemyongpakvn is right that ticking the group in the categories fixes it, while the report is still describing something worth fixing.

Created a group through the same command the back office uses, then asked for the menu tree as each group sees it:

```
new group: category_group rows = 1 of 9 categories   (the home category only)

default customer group -> menu entries under Home: 3 | other accessible categories: 7
the new group          -> menu entries under Home: 0 | other accessible categories: 0
```

`ps_mainmenu` builds its tree with `Category::getNestedCategories($id, $id_lang, false, $this->user_groups)`, which filters on `cg.id_group IN (...)`. A group that has no rows in `ps_category_group` beyond the home category therefore resolves to zero menu entries, so the menu is empty for anyone logged into that group and full for a visitor, who is in the default group. That is exactly the pair of screenshots in the report.

The reason the group starts out that way is #27674: `Group::add()` calls `Category::setNewGroupForHome()` - the home category only - right next to `Carrier::assignGroupToAllCarriers()`, which assigns every carrier. I measured that separately today and posted the detail there, including that the product question the ticket was waiting on was already answered in-thread by @MatShir and @Hlavtox (ask the merchant at creation rather than deciding for them), and that what actually blocks it now is that the group form is still the legacy one.

So I would suggest closing this in favour of #27674, or keeping it as the front-office symptom and fixing it there - either way there is nothing separate to fix in `ps_mainmenu`.

**One caution about the query above, since it is posted on both tickets and people will copy it.** @sitte's version cross-joins every group with every category, so it also grants access to groups that were deliberately restricted - which is the case @leemyongpakvn described on the other ticket, where a group exists precisely so that it should *not* see everything. If you need it, restrict it to the group you just created:

```sql
INSERT INTO ps_category_group (id_category, id_group)
SELECT c.id_category, <id_group> FROM ps_category c
LEFT JOIN ps_category_group cg ON cg.id_category = c.id_category AND cg.id_group = <id_group>
WHERE cg.id_category IS NULL;
```
